app/models/discussion.rb

```diff
@@ -129,32 +136,12 @@ def delayed_destroy
   end
 
   def thread_item_created!(item)
-    self.items_count += 1
-    self.last_item_at = item.created_at
-
-    if SALIENT_ITEM_KINDS.include? item.kind
-      self.salient_items_count += 1
-      self.last_activity_at = item.created_at
-    end
-
-    if item.kind == 'new_comment'
-      self.comments_count += 1
-      self.last_comment_at = item.created_at
-    end
-
-    if self.first_sequence_id == 0
-      self.first_sequence_id = item.sequence_id
-    end
-
+    self.first_sequence_id = item.sequence_id if self.first_sequence_id == 0
     self.last_sequence_id = item.sequence_id
-
     save!(validate: false)
   end
 
   def thread_item_destroyed!(item)
-    self.items_count -= 1
-    self.salient_items_count -= 1 if SALIENT_ITEM_KINDS.include? item.kind
-
     if item.sequence_id == first_sequence_id
```

Comment:
The sequence ids could likely be counter cached too, but didn't bother with that yet; I'm still not 100% sold on this gem.